File: task-manager-ios/task-manager-ios/Data/Remote/Services/TaskRemoteService.swift
//
//  TaskRemoteService.swift
//  task-manager-ios
//
//  Created by Jeofferson Dela Peña on 3/9/24.
//

import Foundation

class TaskRemoteService {
    func getTasks() -> Result<[TaskRemoteDTO], TaskManagerError> {
        // Network logic here...
        return .success([])
    }
    
    func addTask(_ taskRemoteDTO: TaskRemoteDTO) -> Result<Void, TaskManagerError> {
        // Network logic here...
        return .success(())
    }
    
    func editTask(_ taskRemoteDTO: TaskRemoteDTO) -> Result<Void, TaskManagerError> {
        // Network logic here...
        return .success(())
    }
    
    func toggleTaskCompletion(_ taskRemoteDTO: TaskRemoteDTO) -> Result<Void, TaskManagerError> {
        // Network logic here...
        return .success(())
    }
    
    func deleteTask(_ taskRemoteDTO: TaskRemoteDTO) -> Result<Void, TaskManagerError> {
        // Network logic here...
        return .success(())
    }
}

File: task-manager-android/app/src/main/java/com/jeoffersondelapena/task_manager_android/data/remote/service/TaskRemoteService.kt
package com.jeoffersondelapena.task_manager_android.data.remote.service

import com.jeoffersondelapena.task_manager_android.data.remote.dto.TaskRemoteDto
import com.jeoffersondelapena.task_manager_android.domain.util.helper.TaskManagerException
import com.jeoffersondelapena.task_manager_android.domain.util.helper.TaskManagerResult

class TaskRemoteService {
    fun getTasks(): TaskManagerResult<List<TaskRemoteDto>, TaskManagerException> {
        // Network logic here...
        return TaskManagerResult.Success(listOf())
    }

    fun addTask(taskRemoteDto: TaskRemoteDto): TaskManagerResult<Unit, TaskManagerException> {
        // Network logic here...
        return TaskManagerResult.Success(Unit)
    }

    fun editTask(taskRemoteDto: TaskRemoteDto): TaskManagerResult<Unit, TaskManagerException> {
        // Network logic here...
        return TaskManagerResult.Success(Unit)
    }

    fun toggleTaskCompletion(taskRemoteDto: TaskRemoteDto): TaskManagerResult<Unit, TaskManagerException> {
        // Network logic here...
        return TaskManagerResult.Success(Unit)
    }

    fun deleteTask(taskRemoteDto: TaskRemoteDto): TaskManagerResult<Unit, TaskManagerException> {
        // Network logic here...
        return TaskManagerResult.Success(Unit)
    }
}
